code/game/objects/items.dm

```diff
@@ -1323,3 +1392,76 @@ modules/mob/living/carbon/human/life.dm if you die, you will be zoomed out.
 /obj/item/proc/vendor_equip(mob/user)
 	return FALSE
 
+
+/obj/item/proc/color_item(obj/item/facepaint/paint, mob/user)
+
+	if(paint.uses < 1)
+		to_chat(user, span_warning("\the [paint] is out of color!"))
+		return
+
+	var/list/selection_list = list()
+	if(colorable_allowed & COLOR_WHEEL_ALLOWED)
+		selection_list += COLOR_WHEEL
+	if(colorable_allowed & PRESET_COLORS_ALLOWED && length(colorable_colors)>1)
+		selection_list += PRESET_COLORS
+	if(colorable_allowed & ICON_STATE_VARIANTS_ALLOWED && (length(icon_state_variants)>1))
+		selection_list += VARIANTS
+
+	var/selection
+	if(length(selection_list) == 1)
+		selection = selection_list[1]
+	else
+		selection = tgui_input_list(user, "Choose a color setting", name, selection_list)
+
+	if(selection == VARIANTS)
+		var/variant = tgui_input_list(user, "Choose a color.", "Color", icon_state_variants)
+
+		if(!variant)
+			return
+
+		if(!do_after(user, 1 SECONDS, TRUE, src, BUSY_ICON_GENERIC))
+			return
+
+		current_variant = variant
+		update_icon()
+		return
+
+	var/new_color
+	switch(selection)
+		if(PRESET_COLORS)
+			var/color_selection
+			color_selection = tgui_input_list(user, "Pick a color", "Pick color", colorable_colors)
+			if(!color_selection)
+				return
+			if(islist(colorable_colors[color_selection]))
+				var/old_list = colorable_colors[color_selection]
+				color_selection = tgui_input_list(user, "Pick a color", "Pick color", old_list)
+				if(!color_selection)
+					return
+				new_color = old_list[color_selection]
+			else
+				new_color = colorable_colors[color_selection]
+		if(COLOR_WHEEL)
+			new_color = input(user, "Pick a color", "Pick color") as null|color
+
+	if(!new_color || !do_after(user, 1 SECONDS, TRUE, src, BUSY_ICON_GENERIC))
+		return
+
+	set_greyscale_colors(new_color)
+	update_icon()
+
+/obj/item/proc/alternate_color_item(obj/item/facepaint/paint, mob/user)
+	var/list/obj/item/secondaries = list()
+	SEND_SIGNAL(src, COMSIG_ITEM_SECONDARY_COLOR, user, secondaries)
+	if(!length(secondaries))
+		return
```

Comment:
why does this stuff exist if it could be rolled into attackby alternate?

Reply:
Just this part or the whole function? Also this is so that everything can be overwritten by children. Like /clothing needs to update its worn icons when this happens.

Reply:
Whatever cant be bothered to argue